### PR TITLE
feat(wallet): exception handling

### PR DESCRIPTION
**Pull request by zctocm**
1. Component: AccountError
2. Exception handling for a various wallet

---
This change is [Reviewable](https://reviewable.io/reviews/conflux-chain/rigel/69)

### CODE REVIEW

File: src/hooks/useWallet.js [hunk 1]
```diff
@@ -157,3 +160,32 @@ export function useIsNativeToken(chain, tokenAddress) {
...
   )
 }
+
+export function useAccountStatus(chain) {
+  const {address, chainId, error} = useWallet(chain)
+  return useMemo(() => {
+    const wallet = ChainConfig[chain]?.wallet
+    if (wallet) {
+      if (address) {
+        if (isChainIdRight(chain, chainId)) {
```
Comment (angelia-yuqi-personal) [inline src/hooks/useWallet.js:170]: can we use a hook about useIsChainRight?

Comment (zctocm) [reply src/hooks/useWallet.js]: done

File: src/pages/components/AccountError/index.js [hunk 2]
```diff
@@ -0,0 +1,50 @@
...
+      duration: 0,
+    })
+  }
+  return (
+    <>
+      <Tag onClick={onClick} color="error">
+        {text}
+      </Tag>
+    </>
+  )
```
Comment (angelia-yuqi-personal) [inline src/pages/components/AccountError/index.js:44]: we can remove the `<>` `</>` when only have a child `Tag`

File: src/pages/components/AccountError/index.js [hunk 3]
```diff
@@ -0,0 +1,50 @@
...
+      walletName: WalletConfig[chainObject?.wallet]?.name,
+    })
+  } else {
+    text = t('error.index')
+    notiTitle = t('error.connecting')
+    notiContent = t('error.connectingContent')
+  }
+  const onClick = () => {
+    Notification.open({
+      title: `${notiTitle}`,
```
Comment (angelia-yuqi-personal) [inline src/pages/components/AccountError/index.js:32]: can use `notiTitle` directly

File: src/pages/components/AccountError/index.js [hunk 4]
```diff
@@ -0,0 +1,50 @@
...
+      type: 'warning',
+      content: <div>{notiContent}</div>,
```
Comment (angelia-yuqi-personal) [inline src/pages/components/AccountError/index.js:34]: if you want to add some className, we can use `span` as the wrap container,  maybe can use `notiContent` directly here.

File: src/hooks/useWallet.js [hunk 5]
```diff
@@ -157,3 +160,42 @@ export function useIsNativeToken(chain, tokenAddress) {
...
+    }
+  }, [address, chain, error, isChainIdRight])
+}
+
+export function useIsChainIdRight(chain, chainId) {
+  const {wallet, supportedChainIds} = ChainConfig[chain]
+  return useMemo(
+    () =>
+      wallet && chainId == supportedChainIds?.[IS_DEV ? 'TESTNET' : 'MAINNET'],
+    [chainId, supportedChainIds, wallet],
```
Comment (angelia-yuqi-personal) [inline src/hooks/useWallet.js:199]: we can use chainId, chain, IS_DEV, Boolean(wallet) as the dependences here

File: src/utils/index.js [hunk 6]
```diff
@@ -42,3 +42,10 @@ export function calculateGasMargin(value) {
 export function getExponent(decimals) {
   return `1e${decimals}`
 }
+
+// export function isChainIdRight(chain, chainId) {
```
Comment (angelia-yuqi-personal) [inline src/utils/index.js]: remove useless code

Comment (zctocm) [reply src/utils/index.js]: done

File: src/pages/components/AccountError/index.js (hunk 2 above)
Comment (zctocm) [reply src/pages/components/AccountError/index.js:42]: done

File: src/pages/components/AccountError/index.js (hunk 3 above)
Comment (zctocm) [reply src/pages/components/AccountError/index.js]: done

File: src/pages/components/AccountError/index.js (hunk 4 above)
Comment (zctocm) [reply src/pages/components/AccountError/index.js]: done

File: src/hooks/useWallet.js (hunk 5 above)
Comment (zctocm) [reply src/hooks/useWallet.js]: done

**Review by angelia-yuqi-personal — APPROVED** [review]

**Pull request merged**